src/BaGet.GCP/Services/GoogleBucketStorageService.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using BaGet.Core.Services;
+using BaGet.GCP.Configuration;
+using Google;
+using Google.Cloud.Storage.V1;
+using Microsoft.Extensions.Options;
+
+namespace BaGet.GCP.Services
+{
+    public class GoogleBucketStorageService : IStorageService
+    {
+        private readonly string _bucketName;
+
+        public GoogleBucketStorageService(IOptionsSnapshot<GoogleBucketStorageOptions> options)
+        {
+            if (options == null)
+                throw new ArgumentNullException(nameof(options));
+
+            _bucketName = options.Value.BucketName;
+        }
+
+        public async Task<Stream> GetAsync(string path, CancellationToken cancellationToken = default)
+        {
+            var storage = await StorageClient.CreateAsync();
+            var stream = new MemoryStream();
+            await storage.DownloadObjectAsync(_bucketName, path, stream, cancellationToken: cancellationToken);
+            stream.Seek(0, SeekOrigin.Begin);
+            return stream;
+        }
+
+        public Task<Uri> GetDownloadUriAsync(string path, CancellationToken cancellationToken = default)
+        {
+            return Task.FromResult(new Uri(path));
```

Comment:
This returns the incorrect Uri. Two suggestions:

1. Use [`UrlSigner.Sign`](https://www.fuget.org/packages/Google.Cloud.Storage.V1/2.3.0-beta04/lib/netstandard1.3/Google.Cloud.Storage.V1.dll/Google.Cloud.Storage.V1/UrlSigner) to create a signed Uri to the object. This seems hard to do...
2. Assume that the URI is public and return `http://storage.googleapis.com/BUCKET_NAME/OBJECT_NAME`

I would go for the latter.